### PR TITLE
IM-37 Added a route for adding data to the product table.  Also creat…

### PR DESCRIPTION
**Pull request by johnschroeder**
…ed a database convenience module called impdb.js

### CODE REVIEW

File: routes/newProductSubmission.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        )
+    );
+
+
+    connection.connect(function(err) {
+        if (err) {
+            console.error(err.stack);
+            res.status(503).send("Connection Error: " + err.code);
+        } else {
+            connection.beginTransaction(function(err) {
```
Comment (NamespaceValentine) [inline routes/newProductSubmission.js:73]: Awesome, glad to see transactions, and using the convenience method is great so as not to clutter the code with standard (not specific to our needs) MySQL code.

File: routes/newProductSubmission.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
+ var mysql = require("mysql");
+ var config = require("konfig")();
+ var express = require("express");
+ var router = express.Router();
+ /* Is there a way to do this without the relative path? */
+ var db = require("../imp_modules/impdb.js");
```
Comment (NamespaceValentine) [inline routes/newProductSubmission.js]: I will add a module that allows us to remove the need for relative path, but for now this works.

File: imp_modules/impdb.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+/**
```
Comment (NamespaceValentine) [inline imp_modules/impdb.js]: Yay! our first reusable service ^_^.

File: routes/newProductSubmission.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+ /*
+ Usage:
+ localhost:50001/ProductID/Name/Customer/Description/DateCreated
```
Comment (NamespaceValentine) [inline routes/newProductSubmission.js]: usage is actually: 
localhost:50001/newProductSubmission/ProductID/Name/Customer/Description/DateCreated

I would prefer if we could set a standard for what variables vs static stuff in a route looks like:
localhost:50001/newProductSubmission/:ProductID/:Name/:Customer/:Description/:DateCreated
or 
localhost:50001/newProductSubmission/{ProductID}/{Name}/{Customer}/{Description}/{DateCreated}

or something similar. We should also be defining what each variable means in this comment. For examples take a look at:

http://apidocjs.com/

**Comment by NamespaceValentine** [comment]
Other than the usage comment, this all looks good. Will sign off on merge once that is handled, so we can start setting a standard.

**Comment by johnschroeder** [comment]
More work to do

**Pull request closed without merging**